sdk/helper/testcluster/docker/environment.go

```diff
@@ -0,0 +1,946 @@
+		tag = "latest"
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	var seenLogs uberAtomic.Bool
+	logConsumer := func(s string) {
+		if seenLogs.CAS(false, true) {
+			wg.Done()
+		}
+		n.Logger.Trace(s)
+	}
+	logStdout := &LogConsumerWriter{logConsumer}
+	logStderr := &LogConsumerWriter{func(s string) {
+		if seenLogs.CAS(false, true) {
+			wg.Done()
+		}
+		testcluster.JSONLogNoTimestamp(n.Logger, s)
+	}}
+	r, err := dockhelper.NewServiceRunner(dockhelper.RunOptions{
+		ImageRepo: repo,
+		ImageTag:  tag,
+		// We don't need to run update-ca-certificates in the container, because
+		// we're providing the CA in the raft join call, and otherwise Vault
+		// servers don't talk to one another on the API port.
+		Cmd: []string{"server"},
+		Env: []string{
+			// For now we're using disable_mlock, because this is for testing
+			// anyway, and because it prevents us using external plugins.
+			"SKIP_SETCAP=true",
+			"VAULT_LOG_FORMAT=json",
+		},
+		Ports:           []string{"8200/tcp", "8201/tcp"},
+		ContainerName:   n.Name(),
+		NetworkName:     opts.NetworkName,
+		CopyFromTo:      copyFromTo,
+		LogConsumer:     logConsumer,
+		LogStdout:       logStdout,
+		LogStderr:       logStderr,
+		PreDelete:       true,
+		DoNotAutoRemove: true,
+		PostStart: func(containerID string, realIP string) error {
+			err := n.setupCert(realIP)
+			if err != nil {
+				return err
+			}
+
+			// If we signal Vault before it installs its sighup handler, it'll die.
+			wg.Wait()
+			n.Logger.Trace("running poststart", "containerID", containerID, "IP", realIP)
+			return n.runner.RefreshFiles(ctx, containerID)
+		},
+		Capabilities:      []string{"NET_ADMIN"},
+		OmitLogTimestamps: true,
+	})
+	if err != nil {
+		return err
+	}
+	n.runner = r
+
+	svc, _, err := r.StartNewService(ctx, false, false, func(ctx context.Context, host string, port int) (dockhelper.ServiceConfig, error) {
+		config, err := n.apiConfig()
+		if err != nil {
+			return nil, err
+		}
+		config.Address = fmt.Sprintf("https://%s:%d", host, port)
+		client, err := api.NewClient(config)
+		if err != nil {
+			return nil, err
+		}
+		_, err = client.Sys().SealStatus()
+		if err != nil {
+			return nil, err
+		}
+		return dockhelper.NewServiceHostPort(host, port), nil
+	})
+	if err != nil {
+		return err
+	}
+
+	n.HostPort = svc.Config.Address()
+	n.container = svc.Container
+	netName := opts.NetworkName
+	if netName == "" {
+		if len(svc.Container.NetworkSettings.Networks) > 1 {
+			return fmt.Errorf("Set d.RunOptions.NetworkName instead for container with multiple networks: %v", svc.Container.NetworkSettings.Networks)
+		}
+		for netName = range svc.Container.NetworkSettings.Networks {
+			// Networks above is a map; we just need to find the first and
+			// only key of this map (network name). The range handles this
+			// for us, but we need a loop construction in order to use range.
+		}
+	}
+	n.RealAPIAddr = "https://" + svc.Container.NetworkSettings.Networks[netName].IPAddress + ":8200"
```

Comment:
Not 100% sure what you're looking for here, might be simplest to PR what you want, doesn't sound like anything I'd object to.

Reply:
Ah, I see, `dockerClusterNode` is not exported. 

I was hoping for:

```
type dockerClusterNode struct {
    ...

    ContainerNetworkName string
    ContainerIPAddress string
}
```

and where this comment is, adding:

```
n.ContainerNetworkName = netName
n.ContainerIPAddress = svc.Container.NetworkSettings.Networks[netName].IPAddress
```

Reply:
I think exporting this might be the right way to go anyways. As it is, there's no way to directly interact with the Vault container instance if necessary without making this exported, as you wouldn't know the node ID AFAICT? You can call `.Name()` but you'd have to trim the `ClusterName` prefix, which seems a little weird.

Reply:
I don't feel strongly about keeping it private.  Done.